### PR TITLE
feat(op-challenger): Load Onchain Fault Dispute Game Contract Game Depth

### LINKED ISSUE #6677: feat(op-challenger): Load onchain Game Depth
**Description**

When playing a fault dispute game, the `op-challenger` needs to know the game depth for trace index and other game state calculations. Currently, this value is passed into the challenger via a CLI Flag `--game-depth`.

Instead of manually providing this game depth, the challenger should fetch the game depth via the `Loader` fault component on initialization and bubble up the error on failure.

Since the game depth is required for the initialization of the `Agent` fault component, it should be retrieved prior within the service component initialization.

### Tasks
- [ ] Implement a `FetchGameDepth` method on the `Loader` component.
- [ ] Unit test the new `FetchGameDepth` loader component method.
- [ ] On service init, fetch the game depth from the loader and pass this through to the Agent creation.

**Tests**

Unit tests implemented for the `Loader` component.

**Associated Linear Ticket**

CLI-4330

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Loads the onchain fault dispute game game depth
via the loader fault component.

**Tests**

Unit tests surround the loader component.

**Metadata**

Github: #6677
Linear: Fixes CLI-4330

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6678** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6678" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6678?utm_source=stack-comment).

File: op-challenger/flags/flags.go [hunk 1]
```diff
@@ -48,11 +48,6 @@ var (
 		Usage:   "Temporary hardcoded flag if we agree or disagree with the proposed output.",
 		EnvVars: prefixEnvVars("AGREE_WITH_PROPOSED_OUTPUT"),
 	}
-	GameDepthFlag = &cli.IntFlag{
-		Name:    "game-depth",
-		Usage:   "Depth of the game tree.",
-		EnvVars: prefixEnvVars("GAME_DEPTH"),
-	}
```
Comment (ajsutton) [inline op-challenger/flags/flags.go:55]: We should update the `make alphabet` scripts to remove the game-depth arg as well so they keep working.

**Review by ajsutton — APPROVED** [review]
LGTM. Just worth updating the alphabet scripts so they keep working.

File: op-challenger/flags/flags.go (hunk 1 above)
Comment (refcell) [reply op-challenger/flags/flags.go:55]: Believe this is fixed now :)

**Pull request merged**